staging/src/k8s.io/apiserver/pkg/endpoints/filters/tenantInfo.go

```diff
@@ -60,16 +60,16 @@ func WithTenantInfo(handler http.Handler) http.Handler {
 
 // normalizeObjectTenant returns the tenant of the object based on what the user tenant is.
 func normalizeTenant(userTenant, objectTenant string) (string, error) {
-	if userTenant == "" {
+	if userTenant == metav1.TenantNone {
 		return "", fmt.Errorf("The user Tenant is null")
 	}
 
-	if userTenant == "system" {
+	if userTenant == metav1.TenantSystem {
 		// for system user, we don't touch the tenant value in the objectMeta
 		return objectTenant, nil
 	}
 
-	if objectTenant != "" && objectTenant != userTenant {
+	if objectTenant != metav1.TenantNone && objectTenant != userTenant {
```

Comment:
I think we'd better not do the check here.  This should be left to Authorizers.  Otherwise in future it will prevent us from supporting flexible cross-tenant access policies.